bot/__init__.py

```diff
@@ -18,24 +20,84 @@ def case_insensitive_skip_string(self, string: str) -> bool:
     return False
 
 
-def case_insensitive_get_word(self) -> str:
+def _get_word(self) -> str:
     """
     Invokes the get_word method from
     discord.ext.commands.view used to find
     the bot command part of a message, but
-    allows the command to ignore case sensitivity
+    allows the command to ignore case sensitivity,
+    and allows commands to have Python syntax.
+
+    Example of valid Python syntax calls:
+    ------------------------------
+    bot.tags.set("test", 'a dark, dark night')
+    bot.help(tags.delete)
+    bot.hELP(tags.delete)
     """
 
-    word = _get_word(self)
-    if isinstance(word, str):
-        return word.lower()
-    return word
+    pos = 0
+    while not self.eof:
+        try:
+            current = self.buffer[self.index + pos]
+            if current.isspace() or current == "(":
+                break
+            pos += 1
+        except IndexError:
+            break
+
+    self.previous = self.index
+    result = self.buffer[self.index:self.index + pos]
+    self.index += pos
+
+    # get all single and double quote encased args
+    single_quotes = r'[\']([^\']*?)[\']'
+    double_quotes = r'[\"]([^\"]*?)[\"]'
+
+    # start with the type of quote that occurs first
+    buf = self.buffer[self.index:]
+
+    # single quotes occur first
+    if (buf.find("'") < buf.find('"')) and buf.find("'") != -1:
+        first = single_quotes
+        second = double_quotes
+    # double quotes occur first
+    else:
+        first = double_quotes
+        second = single_quotes
+
+    # after we get the first arg, we remove it from the buf so that
+    # quotes inside the arg will not be picked up in the second findall.
+    new_args = re.findall(first, buf)
+    print(new_args)
+    for arg in new_args:
+        buf = buf.replace(arg, "")
+    new_args += re.findall(second, buf)
```

Comment:
In the process of getting myself a staging setup of the bot, but for others' reference while I'm working on that:

This looks like it'll turn the arguments from
```python
bot.command_with_four_args("first", 'second', "third", 'fourth')
```
into
```python
["first", "third", "second", "fourth"]
```

Reply:
by george. you're right. of course you're right. yeah I need to fix that.

Reply:
If that's the case, requesting changes to remove approval. Not sure if we're keeping the debug `print` statements or not but that's another thing I had on the table.

Reply:
dismissing approval. fuck it, I guess we're gonna try with `ast` after all.